src/app/profile/receipts/receipts.component.js

```diff
@@ -0,0 +1,109 @@
+import angular from 'angular';
+import template from './receipts.tpl';
+import donationsService from 'common/services/api/donations.service';
+import filterByYear from './receipts.filter';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import sessionEnforcerService, {EnforcerCallbacks, EnforcerModes} from 'common/services/session/sessionEnforcer.service';
+import {Roles} from 'common/services/session/session.service';
+
+class ReceiptsController {
+
+  /* @ngInject */
+  constructor(donationsService,sessionEnforcerService,$location,$window,$log) {
+    this.donationsService = donationsService;
+    this.sessionEnforcerService = sessionEnforcerService;
+    this.$location = $location;
+    this.$window = $window;
+    this.$log = $log;
+    this.loading = false;
+    this.maxShow = this.step = 25;
+    this.retrievingError = '';
+  }
+
+  $onInit() {
+    this.enforcerId = this.sessionEnforcerService([Roles.registered], {
+      [EnforcerCallbacks.signIn]: () => {
+        this.today = new Date();
+        this.currentYear = this.today.getFullYear();
+        this.getReceipts(this.currentYear,true);
+      },
+      [EnforcerCallbacks.cancel]: () => {
+        this.$window.location = '/';
+      }
+    }, EnforcerModes.donor);
+    this.today = new Date();
+    this.currentYear = this.today.getFullYear();
+    this.getReceipts(this.currentYear,true);
+  }
+
+  getReceipts(year, tryPreviousYear){
+    this.loading = true;
```

Comment:
I think you should put `this.retrievingError = '';` up here so the error will go away after signing in and not still be there while it is loading.